### ISSUE TITLE
Handling Elf.ksy parsing exceptions

### ISSUE THREAD

**Issue by taviso** [kaitai-io/kaitai_struct#754]
Hello, I'm trying to integrate kaitai into my application, if I try to parse `/bin/ls` from Ubuntu 18.04 with `Elf.ksy`, I get an exception. I verified the same thing happens on the Web IDE.

I guess it's not clear to me if I should be handling this error, or if it's a bug?

To reproduce, load the attached binary (It's an original unmodified binary, you could also get it from [here](https://packages.ubuntu.com/bionic/coreutils).

Then use the Web IDE to expand `Elf.header.sectionHeaders[25].body`.

The error is:

```
Parse error: undefined
Call stack: undefined KaitaiEOFError: requested 4832 bytes, but only 2104 bytes available
```

![image](https://user-images.githubusercontent.com/123814/82179277-c5d5ed00-9892-11ea-8c31-1a5478381fc0.png)

**Comment by taviso** [comment]
Oops, github rejected my attachment, lets try this instead.

[sample.elf.gz](https://github.com/kaitai-io/kaitai_struct/files/4642285/sample.elf.gz)

**Comment by GreyCat** [comment]
elf.ksy actually currently fails to compile at all now with this error:

```
elf.ksy:!:!: found duplicate key 1879048193
```

So I assume that you're using old ksc v0.8, which allows these duplicate keys.

In general, if it's a valid source ELF file (and it looks like it's a valid file), then any exception is an indicator of a problem in format specification. You should never get exceptions during parsing if the source file is valid.

So, I would start with:

* Fixing duplicate key problem with elf.ksy
* Retrying with latest ksc
* Potentially, pinpointing and fixing a bug in elf.ksy related to the section that you've demonstrated, using file you've given

**Comment by generalmimon** [comment]
See https://github.com/kaitai-io/kaitai_struct_formats/issues/279

**Comment by generalmimon** [comment]
To put it in perspective, the issue https://github.com/kaitai-io/kaitai_struct_formats/issues/279 relates to the problem with duplicate keys, but I tested the [`sample.elf` file](https://github.com/kaitai-io/kaitai_struct/issues/754#issuecomment-629966818) using [this revision](https://github.com/kaitai-io/kaitai_struct_formats/blob/4bdddf51004cb77f8af86937ff5c1e7f18ec88b1/executable/elf.ksy) of the `executable/elf.ksy` spec and the problem after expanding `Elf.header.sectionHeaders[25].body` still persists.

**Comment by solemnwarning** [comment]
Just a heads up - the bunzip2 executable which is initially open in the web IDE has the same problem. You can see the error in the console, and when trying to export the header to JSON (which doesn't work).

**Comment by armijnhemel** [comment]
Looking at the screenshot: the section has type NOBITS meaning that there is no data associated with it and `size` doesn't describe actual file data.

**Comment by armijnhemel** [comment]
Also see https://github.com/kaitai-io/kaitai_struct_formats/issues/517

**Comment by armijnhemel** [comment]
Should have been fixed now: https://github.com/kaitai-io/kaitai_struct_formats/commit/4d74a6c46c2e3e8e0e592f273648318b1476f020

**Comment by generalmimon** [comment]
> Should have been fixed now: [kaitai-io/kaitai_struct_formats@4d74a6c](https://github.com/kaitai-io/kaitai_struct_formats/commit/4d74a6c46c2e3e8e0e592f273648318b1476f020)

I confirm that the [`sample.elf`](https://github.com/kaitai-io/kaitai_struct/files/4642285/sample.elf.gz) test file from https://github.com/kaitai-io/kaitai_struct/issues/754#issuecomment-629966818 now works => closing.
